File: seed/csharp-model/examples/src/SeedExamplesClient/Commons/Types/EventInfo.cs
using SeedExamplesClient.Commons
using System.Text.Json.Serialization

namespace SeedExamplesClient.Commons

public class EventInfo
{
    public class _Metadata : Metadata
    {
        [JsonPropertyName("type")]
        public string Type { get; } = "metadata";
    }
    public class _Value
    {
        [JsonPropertyName("type")]
        public string Type { get; } = "tag";

        [JsonPropertyName("value")]
        public string Value { get; init; }
    }
}

File: seed/swift-sdk/examples/Sources/Schemas/EventInfo.swift
public enum EventInfo: Codable, Hashable, Sendable {
    case metadata(Metadata)
    case tag(Tag)

    public struct Metadata: Codable, Hashable, Sendable {
        public let type: String = "metadata"
        public let id: String
        public let data: [String: String]?
        public let jsonString: String?
        public let additionalProperties: [String: JSONValue]

        public init(type: String, id: String, data: [String: String]?, jsonString: String?, additionalProperties: [String: JSONValue]) {
        }

        private enum CodingKeys: String, CodingKey {
        }
    }

    public struct Tag: Codable, Hashable, Sendable {
        public let value: Tag

        public init(value: Tag) {
        }

        private enum CodingKeys: String, CodingKey {
        }
    }
}